### ISSUE TITLE
Reduce `match` guard with equality with straightforward `match`

### ISSUE THREAD

**Issue by frewsxcv** [rust-lang/rust-clippy#11697]
### What it does

Prefer matching on the value directly instead of using a guard with equality

### Example

```rust
match x {
    y if y == 100 => ...,
}
```

Could be written as:

```rust
match x {
    100 => ...,
}
```

**Comment by y21** [comment]
This lint already exists: [`redundant_guards`](https://rust-lang.github.io/rust-clippy/master/index.html#/redundant_guards)

Your example code has this warning:
```
warning: redundant guard
 --> src/main.rs:4:14
  |
4 |         y if y == 100 => {},
  |              ^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_guards
  = note: `#[warn(clippy::redundant_guards)]` on by default
help: try
  |
4 -         y if y == 100 => {},
4 +         100 => {},
```

**Comment by frewsxcv** [comment]
This didn't cause a warning for me when using bitflags:

<img width="701" alt="Screenshot 2023-10-22 at 7 53 50 PM" src="https://github.com/rust-lang/rust-clippy/assets/416575/9649afb9-7bd9-4820-90d5-3165e85d4bc5">

**Comment by frewsxcv** [comment]
And yes I have clippy lints turned on in my VSCode

**Comment by y21** [comment]
Looks like a false negative then.
From a quick glance, it looks like clippy doesn't consider a path to a const (`GeomType::POINT`) as [an expression that can be a pattern](https://github.com/rust-lang/rust-clippy/blob/56c8235c99d0138d839b009319416732218dd7ca/clippy_lints/src/matches/redundant_guards.rs#L220-L225), even though it really could be.

Smaller repro that doesn't emit any warnings, without the `bitflags` dep:
```rs
const X: i32 = 1;
match 1 {
    x if x == X => {}
    _ => {}
}
```

**Comment by asquared31415** [comment]
Note that not all consts can be patterns and the precise semantics are being worked out. https://github.com/rust-lang/lang-team/issues/220  Though in this case I would be surprised if anything stopped a const like this from working as a pattern.
